dockstore-client/src/main/java/io/dockstore/client/cli/nested/ToolClient.java

```diff
@@ -710,12 +735,61 @@ private void updateTool(List<String> args) {
+                }
+
+                final String toolMaintainerEmail = optVal(args, "--tool-maintainer-email", tool.getToolMaintainerEmail());
+                final String privateAccess = optVal(args, "--private", tool.getPrivateAccess().toString());
+
+                // Check for correct private access
+                if (!(privateAccess.equalsIgnoreCase("false") || privateAccess.equalsIgnoreCase("true"))) {
+                    errorMessage("The possible values for --private are 'true' and 'false'.", Client.CLIENT_ERROR);
+                }
+
+                // Check that the tool maintainer email is valid
+                if (tool.getMode() == DockstoreTool.ModeEnum.MANUAL_IMAGE_PATH && !toolMaintainerEmail.equals(tool.getToolMaintainerEmail()) && !Strings.isNullOrEmpty(toolMaintainerEmail)) {
+                    EmailValidator emailValidator = EmailValidator.getInstance();
+                    if (!emailValidator.isValid(toolMaintainerEmail)) {
+                        errorMessage("The email address that you entered is invalid.", Client.CLIENT_ERROR);
+                    }
+                }
+
                 tool.setDefaultCwlPath(cwlPath);
                 tool.setDefaultWdlPath(wdlPath);
                 tool.setDefaultDockerfilePath(dockerfilePath);
                 tool.setToolname(toolname);
                 tool.setGitUrl(gitUrl);
 
+                // The following is only for manual tools as only they can be private tools
+                if (tool.getMode() == DockstoreTool.ModeEnum.MANUAL_IMAGE_PATH) {
+                    // Can't set tool maintainer null for private, published repos unless tool author email exists
+                    if (tool.getIsPublished() && tool.getPrivateAccess()) {
+                        if (Strings.isNullOrEmpty(toolMaintainerEmail) && Strings.isNullOrEmpty(tool.getEmail())) {
+                            errorMessage("A published, private tool must have either an tool author email or tool maintainer email set up.",
+                                    Client.CLIENT_ERROR);
+                        }
+                    }
+
+                    tool.setToolMaintainerEmail(toolMaintainerEmail);
+
+                    // When changing public tool to private and the tool is published, either tool author email or tool maintainer email must be set up
+                    if (privateAccess.equalsIgnoreCase("true") && !tool.getPrivateAccess() && tool.getIsPublished()) {
+                        if (Strings.isNullOrEmpty(toolMaintainerEmail) && Strings.isNullOrEmpty(tool.getEmail())) {
+                            errorMessage("A published, private tool must have either an tool author email or tool maintainer email set up.",
+                                    Client.CLIENT_ERROR);
+                        }
+                    }
+
+                    if (privateAccess.equalsIgnoreCase("true")) {
```

Comment:
Ah, ok. Well with your check on line 330, I think this block could just be `tool.setPrivateAccess(privateAccess.equalsIgnoreCase("true"))` Or better yet, just evaluate this to a boolean around like 330 and use that boolean thereafter to avoid using the calls to equalsIgnoreCase